components/org.wso2.carbon.identity.conditional.auth.functions.user/src/main/java/org/wso2/carbon/identity/conditional/auth/functions/user/AssignUserRolesFunctionImpl.java

```diff
@@ -0,0 +1,131 @@
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.wso2.carbon.identity.conditional.auth.functions.user;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.CarbonException;
+import org.wso2.carbon.core.util.AnonymousSessionUtil;
+import org.wso2.carbon.identity.application.authentication.framework.config.model.graph.js.JsAuthenticatedUser;
+import org.wso2.carbon.identity.application.authentication.framework.exception.FrameworkException;
+import org.wso2.carbon.identity.conditional.auth.functions.user.internal.UserFunctionsServiceHolder;
+import org.wso2.carbon.user.core.UserRealm;
+import org.wso2.carbon.user.core.UserStoreException;
+import org.wso2.carbon.user.core.UserStoreManager;
+
+import java.util.List;
+
+/**
+ * Function to update given roles for a given user.
+ * The purpose is to perform role assigning during dynamic authentication.
+ */
+public class AssignUserRolesFunctionImpl implements AssignUserRolesFunction {
+
+    private static final Log LOG = LogFactory.getLog(AssignUserRolesFunctionImpl.class);
+
+    /**
+     * {@inheritDoc}
+     *
+     * @param user           Authenticated user.
+     * @param assigningRoles Roles to be assigned.
+     * @return <code>true</code> If the role assigning is successfully completed. <code>false</code> for any other case.
+     */
+    @Override
+    public boolean assignUserRoles(JsAuthenticatedUser user, List<String> assigningRoles) {
+
+        if (user != null && assigningRoles != null) {
+            try {
+                if (user.getWrapped() != null) {
+                    String tenantDomain = user.getWrapped().getTenantDomain();
+                    String userStoreDomain = user.getWrapped().getUserStoreDomain();
+                    String username = user.getWrapped().getUserName();
+                    UserRealm userRealm = getUserRealm(tenantDomain);
+                    if (userRealm != null) {
+                        UserStoreManager userStore = getUserStoreManager(tenantDomain, userRealm, userStoreDomain);
+                        if (userStore != null) {
+                            userStore.updateRoleListOfUser(
+                                    username,
+                                    new String[0],
+                                    assigningRoles.toArray(new String[0])
+                            );
+                            return true;
+                        } else {
+                            LOG.debug("Null value user store received for the user "
```

Comment:
Change log to "Unable to find userStore for username..." something rather than "Null value ..", include both username and userStoreDomain in the debug log.

instead user.getMember("username"), why not username from line58?